### ISSUE TITLE
Literal not inferred as TypedDict in `yield` statement

### ISSUE THREAD

**Issue by rshin** [microsoft/pyright#2967]
**Describe the bug**
A dictionary literal which matches the specification of a `TypedDict` is not detected as such, if it appears in a `yield` statement of a generator function.

**To Reproduce**
See code below.

**Expected behavior**
No errors reported in the below code.

**Screenshots or Code**
```python
from typing import Generator, TypedDict
class X(TypedDict):
    x: str

# Works fine
a: X = {"x": "x"}

# Works fine
def f1() -> X:
    return {"x": "x"}

# Gives an error
def f2() -> Generator[X, None, None]:
    yield {"x": "x"}

# Works fine
def f3() -> Generator[X, None, None]:
    b: X = {"x": "x"}
    yield b
```

**VS Code extension or command-line**
pyright 1.1.217

**Comment by erictraut** [comment]
Thanks for the bug report. I've added support for bidirectional type inference for yield expressions, so this now works as expected. The change will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.218, which I just published. It will also be included in the next release of pylance.

**Comment by goodmami** [comment]
I see something very similar in pyright 1.1.271 when a literal is in a generator:

```python
from typing import TypedDict

class X(TypedDict):
    x: str

points: list[X] = []
points.extend({"x": c} for c in 'abc')  # "dict[str, str]" is incompatible with "X"
points.extend([{"x": c} for c in 'abc'])  # no problem in list comprehension
```

Shall I open a new issue?

**Comment by erictraut** [comment]
@goodmami, yes open a new issue for this one.
